### PR TITLE
Add user_id to case cycles table

### PR DESCRIPTION
**Pull request by arku**
# 💅🏼 What issue does this fix?

- Adds a `user_id` column to the `users` table.
- Adds associations to the `User` and `CaseCycle` models.
- Update `CaseCycle` factory

Blocked by https://github.com/pieforproviders/pieforproviders/pull/275

## 🍂 Before You Submit

* [ ] Did you write tests?
* [x] Did you run `bundle exec rspec` from the root?
* [x] Did you run `bundle exec rails rswag` from the root?
* [x] Did you run `bundle exec rubocop` from the root?
* [ ] Did you run `yarn lint` in `/client`?
* [ ] Did you run `yarn test` in `/client`?
* [ ] Are your primary keys UUIDs on any new tables?

## 🛷 Deployment Considerations

* [ ] Data Migrations
* [x] Schema Migrations
* [ ] Dependencies

## 🧵 Steps to set up locally

Run `rails d:migrate`

### CODE REVIEW

**Comment by weedySeaDragon** [comment]
Since this isn't linked to an issue that describes _why_ I think you should put that into the description of this PR.
I think that you should copy the text from what @katelovescode  said in Slack [assuming that's ok with her].

File: spec/models/case_cycle_spec.rb [hunk 1]
```diff
@@ -94,8 +94,14 @@
 #  submitted_on    :date             not null
 #  created_at      :datetime         not null
 #  updated_at      :datetime         not null
+#  user_id         :uuid             not null
```
Comment (weedySeaDragon) [inline spec/models/case_cycle_spec.rb:97]: Need to add an example (test) for the `belongs_to` relationship, e.g.
```
it { should belong_to(:user) } # or it { is_expected_to belong_to(:user) }
```

**Review by weedySeaDragon — CHANGES_REQUESTED** [review]
just the 1 thing: need an addition to the _spec

File: spec/models/case_cycle_spec.rb (hunk 1 above)
Comment (katelovescode) [reply spec/models/case_cycle_spec.rb:97]: Concur!

**Review by weedySeaDragon — APPROVED** [review]

**Review by katelovescode — APPROVED** [review]
https://pie-for-providers.slack.com/archives/C017TFWAJLE/p1598919788004600 discussion for posterity

Relevant part: So yes, in the end, I think we should associate it to users, because the user might want to enter a case cycle and then multiple kids to it.

**Pull request merged**